meson.build

```diff
@@ -139,6 +139,14 @@ add_project_link_arguments(
   language: 'c'
 )
 
+# rust is optional
+if add_languages('rust', required: false)
+  have_rust = true
+  add_project_arguments('-DHAVE_RUST', language: 'c')
+else
+  have_rust = false
+endif
```

Comment:
do we want this guarded to a `meson_options.txt` option?

Reply:
I sorta lean on yes as a meson tri-state auto feature where `required` will be controlled by it.  That will let us test builds with either way.

Reply:
Added, good idea.

Reply:
I guess this will fix a lot of the CI.